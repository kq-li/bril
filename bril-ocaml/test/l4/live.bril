# ARGS: live
@main(cond: bool) {
.block1:
  a: int = const 47;
  b: int = const 42;
  br cond .left1 .right;
.left1:
  b: int = const 1;
  c: int = const 5;
  jmp .left2;
.right:
  a: int = const 2;
  c: int = const 10;
  jmp .end;
.left2:
  d: int = add a b;
.end:
  print c;
}
