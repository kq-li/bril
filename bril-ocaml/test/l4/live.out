current calculation: live variables
ins:
	label_block1 -> {cond}
	label_left1 -> {a}
	label_right -> {}
	label_left2 -> {a, b, c}
	label_end -> {c}
outs:
	label_block1 -> {a}
	label_left1 -> {a, b, c}
	label_right -> {c}
	label_left2 -> {c}
	label_end -> {}
